exercises/react/react.ts

```diff
@@ -0,0 +1,102 @@
+//
+// Delete and replace stub with your own implementation
+//
+// Inspired by "How it Works":
+// https://indepth.dev/posts/1269/finding-fine-grained-reactive-programming#how-it-works
+// https://levelup.gitconnected.com/finding-fine-grained-reactive-programming-89741994ddee?source=friends_link&sk=31c66a70c1dce7dd5f3f4229423ad127#4543
+//
+// and "Computations":
+// https://github.com/ryansolid/solid/blob/master/documentation/reactivity.md#user-content-computations
+//
+
+type EqualFn<T> = (lhs: T, rhs: T) => boolean
+type GetterFn<T> = () => T
+type SetterFn<T> = (v: T) => T
+type UnsubscribeFn = () => void
+type UpdateFn<T> = (value?: T) => T
+
+type InputCouple<T> = [GetterFn<T>, SetterFn<T>]
```

Comment:
```suggestion
type InputTuple<T> = [GetterFn<T>, SetterFn<T>]
```

Reply:
Unlike arrays, tuple's have a fixed arity. However the word "tuple" itself doesn't convey any arity. "couple" is a legitimate name for a tuple of two elements - which also sounds like "tuple" (i.e. "tuple" with a specified arity):

https://en.wikipedia.org/wiki/Tuple#Names_for_tuples_of_specific_lengths

A more common moniker is "pair" (which doesn't end with "-ple" - implying "tuple") - so I went with that:

```typescript
type InputPair<T> = [GetterFn<T>, SetterFn<T>]
```

Reply:
I personally like Pair better, but as you point out: Couple is valid. You may choose.